### PR TITLE
230 transcription UI

### LINKED ISSUE #230: Improve transcribing process in UI
**Context:**
As part of #221, @nganthi has improved the transcribing process. See design [here](https://app.zeplin.io/project/5ce554083480691e05af2f74/screen/608bd86820f3fe2fd5d4f67a) (relates to Transcripts section only).

**This ticket is to implement the following features:**
- [x] Add green tick icon next to available transcripts (check font awesome for a good match)
- [x] Show the "uploading progress bar" and offers the ability to cancel the upload. (Also remove the react-fetch-progressbar dependency as no longer in use) [cancelling moved to #248]
- [x] Show the approximate transcribing time (use media duration for now - #240 covers estimation) and offers the ability to cancel the transcribing process [cancelling moved to #248]
- [x] Show the "queued" status if the media is queued by PSTT
- [x] Show the error message if upload or transcribing fails and offers the ability to re-upload or re-transcribe [moved to #244]
- [x] Show file deleted message if media has expired (see https://github.com/bbc/digital-paper-edit-firebase/issues/195)

**Developers' notes:**
- For icons, continue using the Font Awesome library (audio, video, cancel, retry, alert icons)

**Out of scope:**
- Notify the users when their transcription is ready.
- Changing the font

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

Fixes #230 

**Describe what the PR does**    

- [x] Adds a duration field to transcript docs at the time they are created
- [x] Passes duration to the TranscriptRow component in Storybook 

**State whether the PR is ready for review or whether it needs extra work**    

Ready

**Additional context**    

Please review with bbc/digital-paper-edit-storybook/pull/100

### CODE REVIEW

File: src/Components/Workspace/index.js [hunk 1]
```diff
@@ -294,31 +344,20 @@ const WorkspaceView = props => {
   };
 
   const createOrUpdateTranscript = async (item) => {
+    console.log('trying to create a transcript...', item);
```
Comment (allishultes) [inline src/Components/Workspace/index.js:347]:
```suggestion
```

File: src/Components/Workspace/index.js [hunk 2]
```diff
@@ -283,6 +283,56 @@ const WorkspaceView = props => {
...
+    newTranscript.display = true;
+
+    return newTranscript;
+
+  };
+
+  const createOrUpdateWithDuration = async (newTranscript) => {
+    const file = newTranscript.file;
+    const video = document.createElement('video');
+    video.setAttribute('id', 'preload');
```
Comment (allishultes) [inline src/Components/Workspace/index.js:322]:
```suggestion

```

File: src/Components/Workspace/index.js [hunk 3]
```diff
@@ -283,6 +283,55 @@ const WorkspaceView = props => {
 
   // general
 
+  const formatDuration = async (duration) => {
```
Comment (aniablaziak) [inline src/Components/Workspace/index.js:286]: woohoo, this is looking so good now 🤠 

just thinking - maybe we can move this function to the Utils directory? this file is getting huuuge 🙈

**Review by aniablaziak — APPROVED** [review]
Looking good ✨ 

One thing: you might have to pull the latest master before merge, there are a few changes merged into master yesterday that don't show in your branch and could revert these.

File: src/Components/Workspace/index.js (hunk 3 above)
Comment (allishultes) [reply src/Components/Workspace/index.js:286]: Good shout!

**Review by sarahrainbow — APPROVED** [review]
looks great!

**Pull request merged**